File: 02-c/container_example.c
// container_example.c
#include <stdio.h>

struct Container {
   int value;
};

int main() {
   struct Container w1;
   struct Container w2;
   w1.value = 1;
   w2 = w1;
   printf("%i %i\n", w1.value, w2.value);
   w2.value = 2;
   printf("%i %i\n", w1.value, w2.value);
}

File: 02-c-containers/ContainerExample.java
// ContainerExample.java
class Container {
   public int value;
}


public class ContainerExample {

   public static void main (String[] args) {

      Container w1 = new Container();
      Container w2 = new Container();
      w1.value = 1;
      w2 = w1;
      System.out.println(w1.value + " " + w2.value);
      w2.value = 2;
      System.out.println(w1.value + " " + w2.value);
   }
}
